language/move-vm/paranoid-tests/tests/encapsulation_safety/borrow_field.mvir

```diff
@@ -0,0 +1,37 @@
+//# publish
+module 0x2.A {
+  struct C { x: u64 }
+
+  public make(): Self.C {
+  label b0:
+    return C { x: 0};
+  }
+}
+
+//# publish
+module 0x2.B {
+    import 0x2.A;
+    struct C { x: u64 }
+
+    public mut_field() {
+        let v: A.C;
+        let v_ref: &A.C;
+        let x: &mut u64;
+
+    label b0:
+        v = A.make();
+        v_ref = &v;
+        x = &mut move(v_ref).C::x;
```

Comment:
In this case for example, `B` module is trying to borrow field from a resource defined in `A`. That's why we need an extra check for `BorrowField`. cc @sblackshear